Physics3D/geometry/triangleMesh.cpp

```diff
@@ -177,31 +181,82 @@ TriangleMesh::TriangleMesh(UniqueAlignedPointer<float>&& vertices, UniqueAligned
 TriangleMesh::TriangleMesh(int vertexCount, int triangleCount, const Vec3f* vertices, const Triangle* triangles) :
 	MeshPrototype(vertexCount, triangleCount) {
 
-	size_t vertexOffset = getOffset(vertexCount);
+        size_t vertexOffset = 8;//getOffset(vertexCount);
 
+	
+	//printf("vertex_count = ori:%d, now: %lu\ntriangle_count = ori:%d, now: %lu\n", vertexCount, getOffset(vertexCount) * 3, triangleCount, getOffset(triangleCount) * 3),
+	
+	//printf("TriangleMesh::TriangleMesh->vertex_offset = %lu\n", vertexOffset);
+	
+	//printf("TriangleMesh::TriangleMesh->vertexCount = %d\n", vertexCount);
+	
 	float* xValues = this->vertices.get();
 	float* yValues = xValues + vertexOffset;
 	float* zValues = yValues + vertexOffset;
 
+	unsigned int a = 0;
 	for(size_t i = 0; i < vertexCount; i++) {
-		xValues[i] = vertices[i].x;
-		yValues[i] = vertices[i].y;
-		zValues[i] = vertices[i].z;
+	    xValues[a] = vertices[i].x;
+	    yValues[a] = vertices[i].y;
+	    zValues[a] = vertices[i].z;
+	    /*  printf("x = %f\n", xValues[a]);
+	    printf("y = %f\n", yValues[a]);
+	    printf("z = %f\n", zValues[a]);
+	    */a++;
+	    a+=vertexOffset*((a%vertexOffset)==0)*2;
+	    //printf("a = %u\n", a);
+	    
 	}
-	fixFinalBlock(this->vertices.get(), vertexCount);
 
-	size_t triangleOffset = getOffset(triangleCount);
+	unsigned int last_index = a-1;
+	unsigned int size_left =  (((vertexCount + 7) & 0xFFFFFFFFFFFFFFF8)*3-vertexCount*3)/3;
```

Comment:
Lots of math on a single line, split computation over multiple lines with named variables
Also, camelCase names. 
Lastly, the constant `& 0xFFFFFFFFFFFFFFF8` is 64 bits wide. unsigned int is (usually) 32 bits wide. Instead I recommend: `& ~0x7`